### ISSUE TITLE
Assassins getting stuck in weird state

### ISSUE THREAD

**Issue by Heroesflorian** [Sh0wdown/UnofficialCrusaderPatch#632]
Btw have you ever failed an assassin rush against the Rat or other opponents due to the assassins getting stuck in a weird state in a 1 tile wide gap between keep and some wall, after sliding down that wall?

After climbing and sliding down the high wall (for low walls, the issue does not occur), they are standing on the ground, occasionally flickering a bit (animation / visuals), but are fighting units that stand atop the keep (which should not be possible). 
They also cannot be moved away from their location as they ignore any "move here" commands, exactly as they would do while they are climbing or sliding down a wall, despite (visually) standing on the ground and not hanging from a wall.
Moreover, they only deal like less than 3% or so of their regular melee damage and also take (seemingly?) less damage than usual themselves - they still seem to take more damage than they deal, relative to the normal values. 
This lasts until either the assassins die, or any hostile units in attack range die - that is, "any hostile units that would be in attack range if height differences were ignored". 

![paM8uKmmT7](https://user-images.githubusercontent.com/12549394/85479223-0de8dd80-b5be-11ea-87be-b627b179650d.gif)

Here a more reduced scenario... 
![O7Yg8oy02H](https://user-images.githubusercontent.com/12549394/85480087-acc20980-b5bf-11ea-9903-6b6be15ce3ec.gif)

...with a sped-up second half of the "encounter":
![uDW0tFBfLr](https://user-images.githubusercontent.com/12549394/85480358-29ed7e80-b5c0-11ea-9b67-b20b1cdf7df0.gif)

And in comparison, what if the assassin does not climb and slide but instead just stands atop the wall: 
![N7yF7pFJ4S](https://user-images.githubusercontent.com/12549394/85480766-d465a180-b5c0-11ea-8b4d-d71eb28c8618.gif)
The assassin can not attack enemies on the keep, unless the 1 tile gap of low ground is "bridged" with a wall tile, then he can thrust across that 1 tile distance (and vice versa).

**Comment by Panbutt** [comment]
Yeah I've noticed this too, its rather annoying and while Assassins are my favorite unit in the game I tend to rarely use them because they're so bugged.

If you order an assassin to climb a wall/tower/gatehouse/keep and they're interrupted along the way (ie they run into an enemy unit and attack) they forget their previous orders afterwords and just stand there. Meaning you have to keep ordering them to scale castle buildings because they keep forgetting.

**Comment by LordHansCapon** [comment]
A fix for the one tile gap bug AND the assassin "forgetfullness" bug is in testing phase.

**Comment by Heroesflorian** [comment]
Amazing! :)
